### PR TITLE
Minor formatting fixes for the LLVM devdocs

### PR DESCRIPTION
**Pull request by ararslan**
In particular, there were a few places where Julia and GC should be capitalized as well as a couple of typos. I also added language annotation to the code blocks. (Highlighter supports LLVM!)

Addresses some of the things noted by @tkelman in #21888.

Note that CI has been skipped and that it should be removed from the commit message when merging to ensure the docs get built. `make check-whitespace` passed locally.

### CODE REVIEW

**Comment by ararslan** [comment]
Btw @Keno these docs are legendary. Excellent work.

File: doc/src/devdocs/llvm.md [hunk 1]
```diff
@@ -112,36 +112,36 @@ The last step is labor intensive.  Suggestions on a better way would be apprecia
 
 Julia has a generic calling convention for unoptimized code, which looks somewhat
 as follows:
-```
+```c
     jl_value_t *any_unoptimized_call(jl_value_t *, jl_value_t **, int);
```
Comment (tkelman) [inline doc/src/devdocs/llvm.md]: does the indentation do any good if it's already triple-backtick-fenced?

Comment (ararslan) [inline doc/src/devdocs/llvm.md]: I believe they end up showing as indented in that case. Dunno whether that was intentional.

Comment (tkelman) [inline doc/src/devdocs/llvm.md]: if it's just a single line of code, seems unnecessary?

File: doc/src/devdocs/llvm.md [hunk 2]
```diff
@@ -63,29 +63,29 @@ via the environment variable `JULIA_LLVM_ARGS`. Here are example settings using
 ## Debugging LLVM transformations in isolation
 
 On occasion, it can be useful to debug LLVM's transformations in isolation from
-the rest of the julia system, e.g. because reproducing the issue inside julia
+the rest of the Julia system, e.g. because reproducing the issue inside Julia
 would take too long, or because one wants to take advantage of LLVM's tooling
 (e.g. bugpoint). To get unoptimized IR for the entire system iamge, pass the
```
Comment (ajkeller34) [inline doc/src/devdocs/llvm.md]: typo: iamge

File: doc/src/devdocs/llvm.md [hunk 3]
```diff
@@ -63,29 +63,29 @@ via the environment variable `JULIA_LLVM_ARGS`. Here are example settings using
 ## Debugging LLVM transformations in isolation
 
 On occasion, it can be useful to debug LLVM's transformations in isolation from
-the rest of the julia system, e.g. because reproducing the issue inside julia
+the rest of the Julia system, e.g. because reproducing the issue inside Julia
```
Comment (Keno) [inline doc/src/devdocs/llvm.md]: These do actually refer to the executable, so should be lower case.

File: doc/src/devdocs/llvm.md [hunk 4]
```diff
@@ -63,29 +63,29 @@ via the environment variable `JULIA_LLVM_ARGS`. Here are example settings using
...
-entire julia pass-pipeline over the IR. As an example, to generate a system
+entire Julia pass-pipeline over the IR. As an example, to generate a system
 image, one could do:
 ```
 opt -load libjulia.so -julia -o opt.bc unopt.bc
 llc -o sys.o opt.bc
 cc -shared -o sys.so sys.o
 ```
-This system image can then be loaded by julia as usual.
+This system image can then be loaded by Julia as usual.
```
Comment (Keno) [inline doc/src/devdocs/llvm.md]: also lower case here and the one above

**Review by Keno — APPROVED** [review]

**Review by Keno — APPROVED** [review]

**Pull request merged**